satpy/readers/mviri_l1b_fiduceo_nc.py

```diff
@@ -0,0 +1,792 @@
+           advised by [PUG]).
+
+        Note that the timestamps do not increase monotonically with the
+        line number in some cases.
+
+        Returns:
+            Mean scanline acquisition timestamps
+        """
+        # Compute mean timestamp per scanline
+        time = time2d.mean(dim='x')
+
+        # If required, repeat timestamps in y-direction to obtain higher
+        # resolution
+        y = time.coords['y'].values
+        if y.size < target_y.size:
+            reps = target_y.size // y.size
+            y_rep = np.repeat(y, reps)
+            time_hires = time.reindex(y=y_rep)
+            time_hires = time_hires.assign_coords(y=target_y)
+            return time_hires
+        return time
+
+
+class VisQualityControl:
+    """Simple quality control for VIS channel."""
+
+    def __init__(self, mask):
+        """Initialize the quality control."""
+        self._mask = mask
+
+    def check(self):
+        """Check VIS channel quality and issue a warning if it's bad."""
+        use_with_caution = da.bitwise_and(self._mask, 2)
+        if use_with_caution.all():
+            warnings.warn(
+                'All pixels of the VIS channel are flagged as "use with '
+                'caution". Use datasets "quality_pixel_bitmask" and '
+                '"data_quality_bitmask" to find out why.'
+            )
+
+    def mask(self, ds):
+        """Mask VIS pixels with bad quality.
+
+        Pixels are considered bad quality if the "quality_pixel_bitmask" is
+        everything else than 0 (no flag set).
+        """
+        return ds.where(self._mask == 0, np.float32(np.nan))
+
+
+def is_high_resol(resolution):
+    """Identify high resolution channel."""
+    return resolution == HIGH_RESOL
+
+
+class DatasetWrapper:
+    """Helper class for accessing the dataset."""
+
+    def __init__(self, nc):
+        """Wrap the given dataset."""
+        self.nc = nc
+
+    @property
+    def attrs(self):
+        """Exposes dataset attributes."""
+        return self.nc.attrs
+
+    def __getitem__(self, item):
+        """Get a variable from the dataset."""
+        ds = self.nc[item]
+        if self._should_dims_be_renamed(ds):
+            ds = self._rename_dims(ds)
+        elif self._coordinates_not_assigned(ds):
+            ds = self._reassign_coords(ds)
+        self._cleanup_attrs(ds)
+        return ds
+
+    def _should_dims_be_renamed(self, ds):
+        """Determine whether dataset dimensions need to be renamed."""
+        return 'y_ir_wv' in ds.dims or 'y_tie' in ds.dims
+
+    def _rename_dims(self, ds):
+        """Rename dataset dimensions to match satpy's expectations."""
+        new_names = {
+            'y_ir_wv': 'y',
+            'x_ir_wv': 'x',
+            'y_tie': 'y',
+            'x_tie': 'x'
+        }
+        for old_name, new_name in new_names.items():
+            if old_name in ds.dims:
+                ds = ds.rename({old_name: new_name})
```

Comment:
do you need to loop here? doesn't `ds.rename(new_names)` just work?

Reply:
No, unfortunately rename fails if one of the old names is not present in the dataset.